feat(issue #1010 and #1008):Implemented left and right arrow adjustments of timebox and date filters

Vayras opened the pull request:
With This PR Changes:
-Left And Right arrow works correctly and helps you navigate the selected date range from date filters
-Date filters enables you to select a date range
-Test files have been updated
-pagination working correctly and only showing if there are more entries

video size is too big , shared on gdrive
https://drive.google.com/drive/folders/1U36os1AU8tAC0TvnvS0BYt-tipV0NFbM?usp=sharing

Tested on chrome , addresses #1010 #1008

kevkevinpal commented:
looks like two tests are failing 

```

Summary of all failing tests
FAIL src/pages/superadmin/header/SuperAdminHeader.spec.tsx (13.752 s)
  ● Header Component › display header with extras

    expect(received).toBeInTheDocument()

    received value must be an HTMLElement or an SVGElement.
    Received has value: null

      37 |     );
      38 |
    > 39 |     expect(screen.queryByText(hardCodedDateRange)).toBeInTheDocument();
         |                                                    ^
      40 |     expect(screen.queryByText(exportCSVText)).toBeInTheDocument();
      41 |     expect(screen.queryByText(initDateRange)).toBeInTheDocument();
      42 |

      at __EXTERNAL_MATCHER_TRAP__ (node_modules/expect/build/index.js:386:30)
      at Object.throwingMatcher [as toBeInTheDocument] (node_modules/expect/build/index.js:387:15)
      at Object.<anonymous> (src/pages/superadmin/header/SuperAdminHeader.spec.tsx:39:52)

FAIL src/pages/superadmin/statistics/Statistics.spec.tsx
  ● Statistics Component › display about view with extras

    expect(received).toBeInTheDocument()

    received value must be an HTMLElement or an SVGElement.
    Received has value: null

      56 |     }
      57 |     for (let i = 0; i < hardcodedWords.length; i++) {
    > 58 |       expect(screen.queryByText(hardcodedWords[i])).toBeInTheDocument();
         |                                                     ^
      59 |     }
      60 |   });
      61 | });

      at __EXTERNAL_MATCHER_TRAP__ (node_modules/expect/build/index.js:386:30)
      at Object.throwingMatcher [as toBeInTheDocument] (node_modules/expect/build/index.js:387:15)
      at Object.<anonymous> (src/pages/superadmin/statistics/Statistics.spec.tsx:58:53)


Test Suites: 2 failed, 21 passed, 23 total
Tests:       2 failed, 71 passed, 73 total
Snapshots:   0 total
Time:        129.888 s
Ran all test suites.
```

kevkevinpal commented on frontend/app/src/pages/superadmin/statistics/index.tsx:
```diff
@@ -25,7 +26,14 @@ import {
+}
+
+export const Statistics = ({ metrics }: StatisticsProps) => {
+  const a = 1;
+  console.log(a);
+  console.log(metrics, 'asas');
```
I think we can delete this

kevkevinpal commented on frontend/app/src/pages/superadmin/statistics/index.tsx:
```diff
@@ -39,31 +47,31 @@ export const Statistics = () => {
             <img className="logoAlign" src={copygray} alt="" width="27.09px" height="20px" />
             <div>
-              <Title>78</Title>
+              <Title>65</Title>
```
might want to change back to `78` or use value passed in

kevkevinpal commented on frontend/app/src/pages/superadmin/tableComponent/TableComponent.Spec.tsx:
```diff
@@ -96,7 +96,7 @@ it('renders each element in the table in the document', () => {
 it('renders each element in the table in the document', () => {
   const { getByText } = render(<MyTable bounties={mockBounties} />);
-  mockBounties.forEach((bounty) => {
+  mockBounties.forEach((bounty: any) => {
```
you might be able to change this to the `Bounty` type

kevkevinpal commented on frontend/app/src/pages/superadmin/index.tsx:
```diff
@@ -42,27 +55,75 @@ export const SuperAdmin = () => {
+    getMetrics();
+  }, [getMetrics]);
+
+  console.log(bountyMetrics, 'asa');
```
can delete this console.log

Vayras commented:
> FAIL src/pages/superadmin/statistics/Statistics.spec.tsx
>   ● Statistics Component › display about view with extras
> 
>     expect(received).toBeInTheDocument()
> 
>     received value must be an HTMLElement or an SVGElement.
>     Received has value: null
> 
>       56 |     }
>       57 |     for (let i = 0; i < hardcodedWords.length; i++) {
>     > 58 |       expect(screen.queryByText(hardcodedWords[i])).toBeInTheDocument();
>          |                                                     ^
>       59 |     }
>       60 |   });
>       61 | });
> 
>       at __EXTERNAL_MATCHER_TRAP__ (node_modules/expect/build/index.js:386:30)
>       at Object.throwingMatcher [as toBeInTheDocument] (node_modules/expect/build/index.js:387:15)
>       at Object.<anonymous> (src/pages/superadmin/statistics/Statistics.spec.tsx:58:53)

@kevkevinpal I have made some changes to the test files let's see if they pass also addressed the comments

kevkevinpal commented on frontend/app/src/pages/superadmin/statistics/Statistics.spec.tsx:
```diff
@@ -16,34 +16,22 @@ beforeAll(() => {
-      expect(screen.queryByText(hardcodedWords[i])).toBeInTheDocument();
-    }
+  test('display about view with extras', () => {
+    render(<Statistics metrics={mockMetrics} />);
```
I dont see any expect statements we should be still testing for things

lets test for the mockMetrics to show up and specific words like `Bounties, Total Bounties Posted and ect` to be showing up aswell

Vayras replied:
Sure will add those and update this

Vayras commented:
@kevkevinpal made the changes and updated the statistics test file

ecurrencyhodler commented:
Tested on staging and it works. REALLY nice job @Vayras . Paying you out now.
